src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int GROW_FACTOR = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= this.table.length * LOAD_FACTOR) {
+            resize();
+        }
+        int index = calculateIndex(key, table.length);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+            return;
+        } else {
```

Comment:
sorry, maybe I wasn't clear
the point of using return is that we don't have to use `else {}` after that - we can move all the code that goes next one tab left, which is [better practice](https://softwareengineering.stackexchange.com/questions/18454/should-i-return-from-a-function-early-or-use-an-if-statement) overall. If you can, write such guard clauses, instead of bracing lots of code with `{ }`, and creating convoluted maze of if-else constructions